README.md

```diff
@@ -47,26 +47,84 @@ type HttpHandler<'err> = HttpHandler<HttpResponseMessage, 'err>
 
 An `HttpHandler` is a plain function that takes two curried arguments, a `NextFunc` and a `Context`, and returns a new
 `Context` (wrapped in a `Result` and `Task`) when finished. On a high level the `HttpHandler` function takes and returns
-a context object, which means every `HttpHandler` function has full control of the outgoing `HttpRequest` and also the
-resulting response.
+a context object, which means every `HttpHandler` function has full control of the outgoing `Request` and also the
+resulting `Response`.
 
 Each HttpHandler usually adds more info to the `HttpRequest` before passing it further down the pipeline by invoking the
 next `NextFunc` or short circuit the execution by returning a result of `Result<Context<'a>, ResponseError>`. E.g if an
 HttpHandler detects an error, then it can return `Result.Error` to fail the processing.
 
-The easiest way to get your head around a Oryx `HttpHandler` is to think of it as a functional Web request processing
-pipeline. Each handler has the full `Context` at its disposal and can decide whether it wants to fail the request by
-returning an `Error`, or continue the request by passing on a new `Context` to the "next" handler, `NextFunc`.
+The easiest way to get your head around a Oryx `HttpHandler` is to think of it as a functional web request processing
+pipeline. Each handler has the full `Context` at its disposal and can decide whether it wants to fail the request or
+continue the request by passing on a new context to the "next" handler.
 
-The more complex way to think about a `HttpHandler` is that there are in fact 3 different ways it may process the
-request:
+1. Call the next handler `NextFunc` with a result value (`'a`), and return (`return!`) what the next handler is
+   returning. Here you have the option to eliding the await by just synchronously return (`return`) the `Task` returned by the
+   `next` function.
+2. Return an `Error`result to short circuit the processing and fail the request.
+3. It is technically possible to also return `Ok` to short circuit the processing, but this is not something you would normally would do.
```

Comment:
"this is not something you would normally would do." -> "but this is not something you would normally do."

Reply:
Thanks, fixed!